src/assets/styles/header.scss

```diff
@@ -0,0 +1,279 @@
+.header {
+  background: $dark;
+  text-align: center;
+  padding: 1em;
+}
+
+.header-inner {
+  margin: 0 auto;
+  width: 80%;
+  max-width: 500px;
+}
+
+.toolbar {
+  position: relative;
+}
+
+.toolbar.-desktop {
+  display: none;
+}
+
+.logo {
+  max-width: 210px;
+}
+
+.cta {
+  font-family: "Freight Sans CMP";
+  color: $white;
+  font-size: 2em;
+  margin: .1em 0;
+}
+
+.search-wrap {
+  height: 2em;
+  overflow: auto;
+  input,
+  button {
+    height: 100%;
+    display: block;
+    float: left;
+    padding: 0;
+    margin: 0;
+    border: 0;
+    box-sizing: border-box;
+  }
+  input {
+    width: 80%;
+  }
+  button {
+    width: 20%;
+    background: $dark-red;
+    &:hover {
+      background: darken($dark-red, 5%);
+    }
+  }
+}
+
+.button-group {
+  margin-top: $offset;
+  overflow: auto;
+  button {
+    width: calc(50% - #{$offset / 2});
+    float: left;
+    &:last-child {
+      float: right;
+    }
+  }
+}
+
+.filter-events {
+  position: fixed;
+  z-index: 9999;
+  left: 0;
+  right: 0;
+  // NOTE: extreme magic number.
+  top: 82px;
+  bottom: 0;
+  background: $white;
+  text-align: left;
+}
+
+.filter-events-header {
+  background: $blue;
+  padding: .5em 1em;
+  color: $white;
+  font-weight: bold;
+  font-family: "Freight Sans CMP";
+  text-transform: uppercase;
+}
+
+.filter-events-body {
+  padding: 0 1em;
+  > * {
+    margin-top: 1em;
+  }
+}
+
+.filter-events-close {
+  display: block;
+  float: right;
+  &:hover {
+    cursor: pointer;
+  }
+}
+
+.filter-events-search {
+  width: 100%;
+  background: $dark-red;
+}
+
+.filter-events-title {
+  font-size: $text-size;
+  font-weight: bold;
+  font-family: "Freight Sans CMP";
+  text-transform: uppercase;
+  padding-bottom: .5em;
+  border-bottom: solid 2px $light-gray;
+  margin: 0 0 1em 0;
+}
+
+.filter-events-item {
+  margin: 1em 0;
+}
+
+.filter-events-checkbox {
+  position: relative;
+  height: 1em;
+  width: 1em;
+  border: solid 2px $black;
+  margin-right: 1em;
+  display: inline-block;
+  font-size: .8em;
+  margin-right: 1em;
+  font-weight: bold;
+  input {
+    visibility: hidden;
+  }
+  label {
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    &:hover {
+      cursor: pointer;
+    }
+  }
+  input:checked + label {
+    background: $light-blue;
+  }
+}
+
+.filter-events-date-picker {
+  input {
+    border: none;
+    border-bottom: solid 1px $dark;
+    max-width: 33%;
+    padding-left: 0;
+    font-size: .8em;
+  }
+  strong {
+    font-size: .8em;
+    margin: 1em;
+  }
+}
+
+.button-dropdown {
+  position: relative;
+  display: inline-block;
+}
+
+.button-dropdown-button {
+  color: $black;
+  background: $white;
+  border: solid 1px $light-gray;
+  transition: background .3s;
+  &:hover {
+    background: $off-white;
+  }
+  span {
```

Comment:
ahh yeah I've been there. For shapes and stuff I think mixins are good since they get crammed into pseudo elements so often.